### PR TITLE
Feature/amrit s/1044602031 improved accounts + JWT

### PR DESCRIPTION
**Pull request by Amrit-S**
### Administrative Info
Monday Board ID: 904869142
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Added more design to account pages (Login, Register, Reset Password), specifically hover effects and errored out input fields as part of form feedback 
- Fixed bug where yellow border padding was seeping over the NavBar 
- Synced Local Storage JWT with backend verification of JWT via the isAuthenticated() function 
- Refactored Admin.js into a React Hook so the page could be authenticated properly
- Added "loading" states for Admin.js, Register.js, and Login.js while they are authenticating the token via backend fetch call
- Added expiry date to JWT tokens (1 hour currently) for extra level of security (on advice of VP Tech David Hacker)
- Refactored NavBar to not be dependent on props, but rather based on the page's current URL 

### Testing
How did you confirm your changes worked? 
- All changes were visually tested to some degree
- For all account pages, I tried many different types of error combinations to make sure the correct error messages and boxes were being errored out (demo below shows a few of the cases)
- All JWT tokens were tested to confirm that they had access to the current pages, including giving some faulty token to verify that it was rejected by the validation (it was) 
- Played with NavBar to see if all pages were properly highlighted upon clicking 
- Tested expiryDate on JWT by setting it temporarily to 30 seconds, and then logging in and seeing if the site kicked me out after 30 seconds (it did)

_**Note: Verification of admin jwt token (if it exists) via backend fetch call is only made whenever a user switches to a new page in the domain_  

### Confirmation of Change 

https://user-images.githubusercontent.com/42155577/108021404-5896ea00-6fd3-11eb-8b6e-e1ef9a604325.mp4

### CODE REVIEW

File: frontend/src/components/NavBar.js [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
-
-import React from 'react';
 import { useHistory } from "react-router-dom";
 import { Navbar, Nav } from 'react-bootstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { faShoppingCart } from '@fortawesome/free-solid-svg-icons'
 import Logo from "../media/UWEAST_Logo_Detail_Transparent.png";
 import '../css/NavBar.css';
-import { isAuthenticated, logout } from '../util/auth';
+import { isAuthenticated, logout} from '../util/Auth';
```
Comment (PatrickBrown1) [inline frontend/src/components/NavBar.js:14]: Getting an issue here, "Auth" doesn't exist in the files I pulled from this branch (but "auth" does). Maybe you just forgot to rename the auth file in utils. Also getting this error in Login.js, Admin.js, and Register.js

**Review by PatrickBrown1 — APPROVED** [review]
The styling looks great. The conversion to functional component looks good (I even got to learn some new useState tricks from your code). Not too familiar with JWT, but it's pretty clear you know what you're doing and it works when I run it on my end. The only issue I had was the file naming. "auth.js" brought up a couple of issues, it seems like you expected it to be named "Auth.js". I capitalized it on my end and it worked.

Overall great job! I'll approve but make sure to fix that capitalization issue.

**Review by amiteshksharma — APPROVED** [review]
LGTM, no problems with different screen sizes and is compatible with mobile/tablet screen sizes as well. All error handling works well.

**Comment by amiteshksharma** [comment]
> LGTM, no problems with different screen sizes and is compatible with mobile/tablet screen sizes as well. All error handling works well.

Also, small thing I noticed in the backend consoles, I saw: 

OPTIONS /user/resetPassword 204 0.307 ms - 0
{
  email: 'dsa@gmail.com',
  oldPassword: 'dasdsasdas',
  newPassword: 'asdfghj'
}
POST /user/resetPassword 401 36.338 ms - 61
OPTIONS /user/login 204 0.155 ms - 0

Ask Thomas to get rid of them or you could do it, up to you guys I guess.

**Pull request merged**